File: strings/1759_count_number_of_homogenous_substrings.go
func countHomogenous(s string) int {

    MOD := int(math.Pow(10, 9)) + 7
    ret := 0
    count := 1
    prev := 'A'

    for _, i := range s{

        if i == prev {
            count = count + 1
        } else {
            count = 1
        }

        prev = i
        ret = ret + count
    }

    return ret % MOD

}

File: strings/1759_count_number_of_homogenous_substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countHomogenous = function(s) {

    const MOD = 10**9 + 7
    let ret = 0, prev = "A", count = 1

    for(l of s){
        if(l === prev) count ++
        else count = 1

        prev = l
        ret += count

    }

    return ret % MOD

};